### PR TITLE
Polyhedron_demo : Enhance Hole_filling_plugin

### LINKED ISSUE #1707: hole filling should allow to select a border by a selection of edges
Request for enhancement: make the hole filling plugin accept a selection of edges as hole.

### PR DESCRIPTION
**Pull request by maxGimeno**
This PR fixes #1707 . It adds a way to fill a hole delimited by an edge selection.
It links the hole_filling_plugin to the Scene_polyhedron_selection_item and changes the applicable() function to return true in case the selected item is a polyhedron_item OR a selection_item.

### CODE REVIEW

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 1]
```diff
@@ -311,16 +316,17 @@ class Polyhedron_demo_hole_filling_plugin :
     dock_widget->hide();
   }
   Scene_hole_visualizer* get_hole_visualizer(Scene_polyhedron_item* poly_item) {
-      return visualizers[poly_item];
+    return visualizers[poly_item];
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 2]
```diff
@@ -383,14 +389,14 @@ public Q_SLOTS:
 
   static QString no_selected_hole_visualizer_error_message() {
     return "Error: please select a hole visualizer from Geometric Objects list."
-      "Use 'Visualize Holes' button to create one by selecting the polyhedron item!";
+           "Use 'Visualize Holes' button to create one by selecting the polyhedron item!";
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 3]
```diff
@@ -383,14 +389,14 @@ public Q_SLOTS:
...
   }
 
 }; // end Polyhedron_demo_hole_filling_plugin
 
 void Polyhedron_demo_hole_filling_plugin::init(QMainWindow* mainWindow,
-                                      CGAL::Three::Scene_interface* scene_interface,
-                                      Messages_interface* m)
+                                               CGAL::Three::Scene_interface* scene_interface,
+                                               Messages_interface* m)
 {
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 4]
```diff
@@ -440,22 +447,22 @@ void Polyhedron_demo_hole_filling_plugin::item_about_to_be_destroyed(CGAL::Three
   else {
     Scene_hole_visualizer* visu_item = qobject_cast<Scene_hole_visualizer*>(scene_item);
     if(visu_item) {
-        visualizers.remove(visu_item->poly_item);
+      visualizers.remove(visu_item->poly_item);
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 5]
```diff
@@ -440,22 +447,22 @@ void Polyhedron_demo_hole_filling_plugin::item_about_to_be_destroyed(CGAL::Three
...
     }
   }
 }
 // removes Scene_hole_visualizer items
 void Polyhedron_demo_hole_filling_plugin::dock_widget_closed() {
   // remove all Scene_hole_visualizer items
   for(CGAL::Three::Scene_interface::Item_id i = 0, end = scene->numberOfEntries();
-    i < end; ++i)
+      i < end; ++i)
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 6]
```diff
@@ -501,13 +508,13 @@ void Polyhedron_demo_hole_filling_plugin::on_Fill_selected_holes_button() {
...
   int counter = 0;
   int filled_counter = 0;
   for(Scene_hole_visualizer::Selected_holes_set::iterator it = hole_visualizer->selected_holes.begin();
-    it != hole_visualizer->selected_holes.end(); ++it, ++counter) {
-      print_message(tr("Hole %1:").arg(counter));
-      if( fill(*(hole_visualizer->poly_item->polyhedron()), (*it)->halfedge) ) { ++filled_counter;}
+      it != hole_visualizer->selected_holes.end(); ++it, ++counter) {
+    print_message(tr("Hole %1:").arg(counter));
+    if( fill(*(hole_visualizer->poly_item->polyhedron()), (*it)->halfedge) ) { ++filled_counter;}
   }
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 7]
```diff
@@ -539,7 +546,7 @@ void Polyhedron_demo_hole_filling_plugin::on_Select_small_holes_button() {
   std::size_t threshold = ui_widget.vertices_threshold_spin_box->value();
   typedef Scene_hole_visualizer::Polyline_data_list::const_iterator const_iterator;
   for(const_iterator it = hole_visualizer->polyline_data_list.begin();
-                     it != hole_visualizer->polyline_data_list.end(); ++it)
+      it != hole_visualizer->polyline_data_list.end(); ++it)
   {
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 8]
```diff
@@ -569,18 +576,18 @@ void Polyhedron_demo_hole_filling_plugin::on_Create_polyline_items_button(){
...
-    it != hole_visualizer->selected_holes.end(); ++it) {
-      Scene_polylines_item* polyline_item = new Scene_polylines_item();
-      polyline_item->polylines = (*it)->polyline->polylines;
-      polyline_item->setName(QString("selected hole %1").arg(counter++));
-      scene->addItem(polyline_item);
+      it != hole_visualizer->selected_holes.end(); ++it) {
+    Scene_polylines_item* polyline_item = new Scene_polylines_item();
+    polyline_item->polylines = (*it)->polyline->polylines;
+    polyline_item->setName(QString("selected hole %1").arg(counter++));
+    scene->addItem(polyline_item);
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 9]
```diff
@@ -619,35 +626,35 @@ bool Polyhedron_demo_hole_filling_plugin::fill
   std::vector<Polyhedron::Facet_handle> patch;
   if(action_index == 0) {
     CGAL::Polygon_mesh_processing::triangulate_hole(poly,
-             it, std::back_inserter(patch),
-             CGAL::Polygon_mesh_processing::parameters::use_delaunay_triangulation(use_DT));
+                                                    it, std::back_inserter(patch),
+                                                    CGAL::Polygon_mesh_processing::parameters::use_delaunay_triangulation(use_DT));
   }
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 10]
```diff
@@ -619,35 +626,35 @@ bool Polyhedron_demo_hole_filling_plugin::fill
...
   else if(action_index == 1) {
     CGAL::Polygon_mesh_processing::triangulate_and_refine_hole(poly,
-             it, std::back_inserter(patch), CGAL::Emptyset_iterator(),
-             CGAL::Polygon_mesh_processing::parameters::density_control_factor(alpha).
-             use_delaunay_triangulation(use_DT));
+                                                               it, std::back_inserter(patch), CGAL::Emptyset_iterator(),
+                                                               CGAL::Polygon_mesh_processing::parameters::density_control_factor(alpha).
+                                                               use_delaunay_triangulation(use_DT));
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 11]
```diff
@@ -619,35 +626,35 @@ bool Polyhedron_demo_hole_filling_plugin::fill
...
-                (CGAL::internal::Uniform_weight_fairing<Polyhedron>(poly)).
-              density_control_factor(alpha).
-              fairing_continuity(continuity).
-              use_delaunay_triangulation(use_DT)));
+                                                                                                    it, std::back_inserter(patch), CGAL::Emptyset_iterator(),
+                                                                                                    CGAL::Polygon_mesh_processing::parameters::weight_calculator
+                                                                                                    (CGAL::internal::Uniform_weight_fairing<Polyhedron>(poly)).
+                                                                                                    density_control_factor(alpha).
+                                                                                                    fairing_continuity(continuity).
+                                                                                                    use_delaunay_triangulation(use_DT)));
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 12]
```diff
@@ -619,35 +626,35 @@ bool Polyhedron_demo_hole_filling_plugin::fill
...
-              it, std::back_inserter(patch), CGAL::Emptyset_iterator(),
-              CGAL::Polygon_mesh_processing::parameters::weight_calculator(CGAL::internal::Cotangent_weight_with_voronoi_area_fairing<Polyhedron>(poly)).
-              density_control_factor(alpha).
-              fairing_continuity(continuity).
-              use_delaunay_triangulation(use_DT)));
+                                                                                                    it, std::back_inserter(patch), CGAL::Emptyset_iterator(),
+                                                                                                    CGAL::Polygon_mesh_processing::parameters::weight_calculator(CGAL::internal::Cotangent_weight_with_voronoi_area_fairing<Polyhedron>(poly)).
+                                                                                                    density_control_factor(alpha).
+                                                                                                    fairing_continuity(continuity).
+                                                                                                    use_delaunay_triangulation(use_DT)));
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 13]
```diff
@@ -656,7 +663,7 @@ bool Polyhedron_demo_hole_filling_plugin::fill
 
   if(patch.empty()) {
     print_message(tr("Warning: generating patch is not successful! %1")
-      .arg(use_DT ? "Please try without 'Use 3D Delaunay Triangulation'!" : ""));
+                  .arg(use_DT ? "Please try without 'Use 3D Delaunay Triangulation'!" : ""));
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 14]
```diff
@@ -667,17 +674,17 @@ bool Polyhedron_demo_hole_filling_plugin::fill
...
     // this part might need speed-up
     bool intersected = false;
-    for(Intersected_facets::iterator it = intersected_facets.begin(); 
-      it != intersected_facets.end() && !intersected; ++it) {
-      for(std::vector<Polyhedron::Facet_handle>::iterator it_patch = patch.begin(); 
-        it_patch != patch.end() && !intersected; ++it_patch) {
+    for(Intersected_facets::iterator it = intersected_facets.begin();
+        it != intersected_facets.end() && !intersected; ++it) {
+      for(std::vector<Polyhedron::Facet_handle>::iterator it_patch = patch.begin();
+          it_patch != patch.end() && !intersected; ++it_patch) {
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp [hunk 15]
```diff
@@ -667,17 +674,17 @@ bool Polyhedron_demo_hole_filling_plugin::fill
     typedef std::vector<std::pair<Polyhedron::Facet_const_handle, Polyhedron::Facet_const_handle> > Intersected_facets;
     Intersected_facets intersected_facets;
     CGAL::Polygon_mesh_processing::self_intersections(poly,
-      std::back_inserter(intersected_facets),
-      CGAL::Polygon_mesh_processing::parameters::vertex_point_map(get(CGAL::vertex_point, poly)));
+                                                      std::back_inserter(intersected_facets),
+                                                      CGAL::Polygon_mesh_processing::parameters::vertex_point_map(get(CGAL::vertex_point, poly)));
```
Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp]: no need to change indentation

**Comment by maxGimeno** [comment]
Actually all of this is an accident. I didn't mean to re-indent the whole code, I guess I missed a selection when I performed a Ctrl+I to re-indent a new piece of code.

**Pull request merged**